sql/mdl.cc

```diff
@@ -1091,6 +1225,16 @@ uint MDL_ticket::get_deadlock_weight() const
          DEADLOCK_WEIGHT_DDL : DEADLOCK_WEIGHT_DML;
 }
 
+void MDL_ticket::get_wait_info(char *output, size_t *output_length) {
+  char buffer[256];
```

Comment:
You do not need `buffer`, can write directly to `output`, if you pass the size of the buffer.

Reply:
Actually, maybe you just return std::string from it?